pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -568,6 +614,39 @@ func manageResourcesReplacement(c *apiserver.APIClient, factories []customresour
 
 // Run runs the KSM check
 func (k *KSMCheck) Run() error {
+	var client *apiserver.APIClient
+
+	// Perform resource discovery if it hasn't successfully completed yet
+	if k.resourceDiscoveryInfo == nil {
+		// Acquire the API Server client, retrying with exponential backoff
+		apiCtx, apiCancel := context.WithTimeout(context.Background(), maximumWaitForAPIServer)
+		defer apiCancel()
+
+		var err error
+		// WaitForAPIClient handles its own retries
+		client, err = apiserver.WaitForAPIClient(apiCtx)
+		if err != nil {
+			return err
```

Comment:
As this is the error displayed in `Status` you may want to handle that differently. The `Retrier` is output most of them time:
```
temporary failure in kubernetes_state_core_resource_discovery, will retry later: try delay not elapsed yet
```

Which is not important. It's better to return the `LastTryError` with some wrapping.